source/postprocess/visualization.cc

```diff
@@ -521,7 +521,13 @@ namespace aspect
               write_master_files (data_out, solution_file_prefix, filenames);
             }
 
-          const unsigned int my_file_id = (group_files == 0) ? my_id : my_id % group_files;
+          const unsigned int n_processes = Utilities::MPI::n_mpi_processes(this->get_mpi_communicator());
+
+          const unsigned int my_file_id = ((group_files == 0) || (group_files >= n_processes)
```

Comment:
This additional check is not needed because the mod doesn't change anything in that case. Do you find that easier to understand? (I don't)